src/main/java/org/broadinstitute/hellbender/tools/walkers/vqsr/scalable/ScoreVariantAnnotations.java

```diff
@@ -0,0 +1,625 @@
+        final int numAll = allAnnotations.length;
+        final List<Double> allScores = new ArrayList<>(Collections.nCopies(numAll, Double.NaN));
+        if (variantTypesToExtract.contains(VariantType.SNP)) {
+            logger.info("Scoring SNP variants...");
+            scoreVariantTypeAndSetElementsOfAllScores(annotationNames, allAnnotations, isSNP, snpScorer, allScores);
+        }
+        if (variantTypesToExtract.contains(VariantType.INDEL)) {
+            logger.info("Scoring INDEL variants...");
+            final List<Boolean> isIndel = isSNP.stream().map(x -> !x).collect(Collectors.toList());
+            scoreVariantTypeAndSetElementsOfAllScores(annotationNames, allAnnotations, isIndel, indelScorer, allScores);
+        }
+        VariantAnnotationsScorer.writeScores(outputScoresFile, Doubles.toArray(allScores));
+        logger.info(String.format("Scores written to %s.", outputScoresFile.getAbsolutePath()));
+    }
+
+    private static void scoreVariantTypeAndSetElementsOfAllScores(final List<String> annotationNames,
+                                                                  final double[][] allAnnotations,
+                                                                  final List<Boolean> isVariantType,
+                                                                  final VariantAnnotationsScorer variantTypeScorer,
+                                                                  final List<Double> allScores) {
+        final File variantTypeAnnotationsFile = LabeledVariantAnnotationsData.subsetAnnotationsToTemporaryFile(annotationNames, allAnnotations, isVariantType);
+        final File variantTypeScoresFile = IOUtils.createTempFile("temp", ".scores.hdf5");
+        variantTypeScorer.score(variantTypeAnnotationsFile, variantTypeScoresFile); // TODO we do not fail until here in the case of mismatched annotation names; we could fail earlier
+        final double[] variantTypeScores = VariantAnnotationsScorer.readScores(variantTypeScoresFile);
+        final Iterator<Double> variantTypeScoresIterator = Arrays.stream(variantTypeScores).iterator();
+        IntStream.range(0, allScores.size()).filter(isVariantType::get).forEach(i -> allScores.set(i, variantTypeScoresIterator.next()));
+    }
+
+    @Override
+    void writeExtractedVariantToVCF(final VariantContext vc,
+                                    final List<Allele> altAlleles,
+                                    final Set<String> labels) {
+        final VariantContextBuilder builder = new VariantContextBuilder(vc);
+        labels.forEach(l -> builder.attribute(l, true)); // labels should already be sorted as a TreeSet
+
+        final List<Double> scores = useASAnnotations
+                ? altAlleles.stream().map(a -> scoresIterator.next()).collect(Collectors.toList())
+                : Collections.singletonList(scoresIterator.next());
+        final double score = Collections.max(scores);
```

Comment:
Yeah, this is just borrowed from ApplyVQSR. It would be nice to retain allele-level scores, but I'm not sure users would make use of them downstream. Things might also get a bit messy for non-multiallelic sites if we have annotations for both site-level and allele-level scores in the mix. I'll add an issue.